### ISSUE TITLE
ERROR Error: Uncaught (in promise): UnknownErrorException: Window is not defined while displaying pdf using base64 string from server

### ISSUE THREAD

**Issue by chetan592** [stephanrauh/ngx-extended-pdf-viewer#1059]
I am using ngx-extended-pdf-viewer to show pdf using base64 strings that I receive from server.

I am not sure but I think it was not there when I was trying it for images from asset folder

For some pdf I get this issue where pdf is not visible and throws error,  which is in console . 

This issue may be because of base library but little direction towards solving it would be very helpful for me.

Adding screenshot for ref

![431FB317-B7E5-4B18-B44D-64031E3DC1AD](https://user-images.githubusercontent.com/47034555/144459321-70081a35-da1d-425b-8fcd-c72b5b1d2a15.jpeg)
![0EFAFE03-1814-475B-91C3-D64AE6B34DFF](https://user-images.githubusercontent.com/47034555/144459334-2e1e6d01-8891-4766-84e5-72011af10e02.jpeg)

**Comment by stephanrauh** [comment]
That may be bad news because it might mean I have to remove a popular feature. So let's examine your case thoroughly first. Which version of ngx-pdf-extended-pdf-viewer are you using? And are you using the default branch or the "bleeding edge" branch?

**Comment by chetan592** [comment]
I am using following version of ngx-extended-pdf-viewer 
"ngx-extended-pdf-viewer": "^11.0.0-alpha.4"

I am on default branch.

Which feature is related with it and what is causing this issue?

**Comment by stephanrauh** [comment]
It's the feature we've decided to build in #804: allow users to filter log messages.

The idea was to replace every console.log call by a custom call. However, the service worker thread doesn't have a Window object, and even trying to check if the Window object exists causes the service worker to crash. I don't understand this. That's not the usual behavior of JavaScript. I'm running out of ideas. There must be a way to detect whether the code is running in a server worker or in a regular UI thread, but I haven't found out yet how to do this!

```typescript
function info(msg) {
  if (verbosity >= VerbosityLevel.INFOS) {
    if (typeof WorkerGlobalScope !== "undefined" && self instanceof WorkerGlobalScope) {
      console.log(`Info: ${msg}`);
    } else if (Window && Window['ngxConsole']) {   <<<<<<< Service worker crashes here!
      Window['ngxConsole'].log(`Info: ${msg}`);
    } else {
      console.log(`Info: ${msg}`);
    }
  }
}
```

**Comment by stephanrauh** [comment]
@chetan592 Please update to version 11.0.0-alpha.10. If this fixes your bug, please tell me. If it doesn't, please tell me, too. I've tried to implement a bugfix in version 11.0.0-alpha.6, and I'd like to know for sure if it fixes the bug or not.

**Comment by chetan592** [comment]
@stephanrauh Thanks for support you are providing to community. 

I updated it to the 11.0.0-alpha.10. and its resolved. 

But there are couple of warnings showing on console. Attached screenshots for your reference.
Anything to worry about them ?

![image](https://user-images.githubusercontent.com/47034555/144700443-0215c799-79fa-4d7b-a71b-35e29fbe2078.png)

**Comment by stephanrauh** [comment]
Hooray! That's good news. But you're right, some of these warnings are exceptions that must be fixed. They prevent displaying the digital signature of the file.

As for the other messages:

- You didn't provide a height attribute, or you passed something like `height="100%"`. More often that not, this results in a height of zero pixels. The PDF viewer tries to fix this. If you're fine with the automated fix, you can ignore the warning. You can also overwrite the global `ngxConsole` function to filter this message. I'm just afraid I haven't documented this feature yet.
- The translation key is a new field of the find bar (the toolbar that shows when you hit the magnifying glass symbol). You'll spot the English text among the translated texts. If you don't mind, send me the translation of your language. I only speak a few languages, so I need the community support to do it right.
- The "Filter: Crypt" message puzzles me. I don't think it's a good sign, but I've never seen it before.
- Digital signatures are shown (at least if you activate the feature), but the PDF viewer can't verify if they are valid or fraud. I want developers to know what they're doing, so I've added a console message.
- You can safely ignore the "undefined function 32". As far as I know, it's a bug of your PDF file. The embedded font is broken. More details here: https://github.com/mozilla/pdf.js/issues/3768#issuecomment-36468349

**Comment by stephanrauh** [comment]
fixed the version 11.0.0-alpha.12.
